src/gt4py/next/program_processors/runners/dace/workflow/translation.py

```diff
@@ -40,15 +41,23 @@ def find_constant_symbols(
         # Search the stride symbols corresponding to the horizontal dimension
         for p in ir.params:
             if isinstance(p.type, ts.FieldType):
-                dims = p.type.dims
-                if len(dims) == 0:
+                h_dims = [dim for dim in p.type.dims if dim.kind == common.DimensionKind.HORIZONTAL]
+                if len(h_dims) == 0:
                     continue
-                elif len(dims) == 1:
-                    dim_index = 0
-                elif len(dims) == 2:
-                    dim_index = 0 if dims[0].kind == common.DimensionKind.HORIZONTAL else 1
+                elif len(h_dims) > 1:
+                    warnings.warn(
+                        f"Unsupported field with multiple horizontal dimensions '{p}'.",
```

Comment:
Maybe throwing an error is not a bad idea since this is something unsupported. It can force the user to fix their code instead of keeping something wrong (even if it accidentally works)